### PR TITLE
adding coverage report for unit tests and a codecov workflow

### PR DESCRIPTION
**Pull request by s-radyuk**
EES-4129
In order to monitor a code quality in the project, we need a codecov integration.

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -38,6 +38,26 @@ jobs:
...
+  unit-test:
+    name: Unit test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Unit tests
+        run: |
+          go mod vendor
```
Comment (vgarvardt) [inline .github/workflows/testing.yml:50]: tests runner should install missing deps automatically, no need to vendor them in advance

**Comment by vgarvardt** [comment]
👍

**Comment by RTodorov** [comment]
+1

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -38,6 +38,24 @@ jobs:
           command: lint
           config: .github/ct.yaml
 
+  unit-test:
```
Comment (aleksandrzhiliaev) [inline .github/workflows/testing.yml:41]: any reason of adding this job?
https://github.com/hellofresh/kangal/pull/24/files#diff-9d29531f011d792c2f2f2c87ac68865fR60
already provides you coverage

Comment (aleksandrzhiliaev) [reply .github/workflows/testing.yml:41]: Example: https://github.com/hellofresh/kangal/runs/1166481484?check_suite_focus=true#step:7:15

**Comment by aleksandrzhiliaev** [comment]
+1

**Pull request merged**